### DISCUSSION TITLE
How to query last inserted item?

### DISCUSSION THREAD

**Discussion by meightythree** [tywalch/electrodb#340] [Q&A] [answered]
I have the following entity and I would like to query the last inserted item, when we don't particularly know the primary key.

```
import { Entity, EntityItem } from 'electrodb';
import { Table } from 'sst/node/table';

import { dynamoDBClient } from 'core/db';

export const MyEntity = new Entity(
    {
        model: {
            entity: 'myEntity',
            version: '1',
            service: 'myService',
        },
        attributes: {
            dateId: {
                type: 'string',
                required: true,
                readOnly: true,
            },
            base: {
                type: 'string',
                readOnly: true,
                required: true,
            },
           // other attributes...
        },
        indexes: {
            myEntity: {
                pk: {
                    field: 'dateId',
                    composite: ['dateId'],
                },
            },
        },
    },
    {
        client: dynamoDBClient,
        table: Table.myEntity.tableName,
    }
);
```

Scan doesn't work for me: `await MyEntity.scan.params({ limit: 1, order: 'desc' }).go()`, it throws `Error: MyEntity.find(...).params(...).go is not a function`

**Comment by tywalch** [comment, marked as answer]
Hi @meightythree 👋 

Two things here:

## Thing 1
Your syntax is nearly correct, the correct syntax would be the following:

```typescript
await MyEntity.scan.go({ limit: 1, order: 'desc' });
```

The `params` and `go` methods are two ways to terminate an operation (terminal methods). The `params` terminal method synchronously returns DynamoDB JSON parameters for a given operation, while the `go` terminal method executes the operation. In the above example I would either call `params` or `go` but not both. You can find more information about those methods here: https://electrodb.dev/en/core-concepts/executing-queries/#execution-methods

## Thing 2
I'm glad you shared your full intent because I don't believe this query will actually get you what you need 😱 Unfortunately it is a DynamoDB thing (not an ElectroDB thing), but scan actually does not guarantee any return order for the items in response 👎 Order is only guaranteed when you query within a partition, and only if your table has a sort key.

Easiest way to accomplish this would be to change your model and table schema to be something like the below model, but if you expect high throughput you may need to consider a more robust approach to prevent hot partitions:

```typescript
import { Entity, EntityItem } from 'electrodb';
import { Table } from 'sst/node/table';

import { dynamoDBClient } from 'core/db';

export const MyEntity = new Entity(
    {
        model: {
            entity: 'myEntity',
            version: '1',
            service: 'myService',
        },
        attributes: {
            dateId: {
                type: 'string',
                required: true,
                readOnly: true,
            },
            base: {
                type: 'string',
                readOnly: true,
                required: true,
            },
           // other attributes...
        },
        indexes: {
            myEntity: {
                sk: {
                    field: 'sk',
                    composite: ['dateId'],
                },
                pk: {
                    field: 'pk',
                    composite: [],
                },
            },
        },
    },
    {
        client: dynamoDBClient,
        table: Table.myEntity.tableName,
    }
);
```
